# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1205291584 14400
#      Tue Mar 11 23:13:04 2008 -0400
# Branch release-3-0-x
# Node ID e13219eaf388da734744b16d1253d340a60e6751
# Parent  7817fd1f2237436c054d10cb1e279e79abd8d6d5
__go_draw_axes__: use strcmpi for text properties; use get for hidden properties

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,10 +1,13 @@
 2008-03-11  John W. Eaton  <jwe@octave.org>
 
+	* plot/__go_draw_axes__.m: Use get to access hidden properties.
+	Use strcmpi when comparing string properties.
+
 	* plot/ChangeLog: Handle axes linewidth property.
 
 	* plot/__go_draw_axes__.m (get_old_gnuplot_color): New subfunction.
 	Use it to replace repeated if/elseif blocks.
 
 2008-03-11  Kai Habel  <kai.habel@gmx.de>
 
         * plot/__go_draw_axes__.m: Plot surfaces in front of axes.
diff --git a/scripts/plot/__go_draw_axes__.m b/scripts/plot/__go_draw_axes__.m
--- a/scripts/plot/__go_draw_axes__.m
+++ b/scripts/plot/__go_draw_axes__.m
@@ -28,35 +28,35 @@ function __go_draw_axes__ (h, plot_strea
 
     parent_figure_obj = get (axis_obj.parent);
 
     persistent have_newer_gnuplot ...
       = compare_versions (__gnuplot_version__ (), "4.0", ">");
 
     ## Set axis properties here?
     pos = [0, 0, 1, 1];
-    if (strcmp (axis_obj.activepositionproperty, "outerposition"))
+    if (strcmpi (axis_obj.activepositionproperty, "outerposition"))
       ymirror = true;
       if (! isempty (axis_obj.outerposition))
 	pos = axis_obj.outerposition;
       endif
     else
       ymirror = false;
       if (! isempty (axis_obj.position))
 	pos = axis_obj.position;
 	fprintf (plot_stream, "set tmargin 0;\n");
 	fprintf (plot_stream, "set bmargin 0;\n");
 	fprintf (plot_stream, "set lmargin 0;\n");
 	fprintf (plot_stream, "set rmargin 0;\n");
       endif
     endif
 
-    if (! strcmp (axis_obj.__colorbar__, "none"))
+    if (! strcmpi (get (h, "__colorbar__"), "none"))
       [pos, cbox_orient, cbox_size, cbox_origin, cbox_mirror] = ...
-	  gnuplot_postion_colorbox (pos, axis_obj.__colorbar__);
+	  gnuplot_postion_colorbox (pos, get (h, "__colorbar__"));
     endif
 
     fprintf (plot_stream, "set origin %.15g, %.15g;\n", pos(1), pos(2));
     fprintf (plot_stream, "set size %.15g, %.15g;\n", pos(3), pos(4));
 
     if (strcmpi (axis_obj.dataaspectratiomode, "manual"))
       r = axis_obj.dataaspectratio;
       fprintf (plot_stream, "set size ratio %.15g;\n", -r(2)/r(1));
@@ -316,34 +316,34 @@ function __go_draw_axes__ (h, plot_strea
     clim = axis_obj.clim;
 
     if (! cautoscale && clim(1) == clim(2))
       clim(2)++;
     endif
 
     [view_cmd, view_fcn, view_zoom] = image_viewer ();
     use_gnuplot_for_images = (ischar (view_fcn)
-			      && strcmp (view_fcn, "gnuplot_internal"));
+			      && strcmpi (view_fcn, "gnuplot_internal"));
 
     ximg_data = {};
     ximg_data_idx = 0;
 
     for i = 1:length (kids)
 
       obj = get (kids(i));
 
       switch (obj.type)
 	case "image"
 	  img_data = obj.cdata;
 	  img_xdata = obj.xdata;
 	  img_ydata = obj.ydata;
 
 	  if (use_gnuplot_for_images)
 
-	    if (strcmp (obj.cdatamapping, "direct"))
+	    if (strcmpi (obj.cdatamapping, "direct"))
 	      cdatadirect = true;
 	    endif
 	    fputs (plot_stream, "set border front;\n");
 	    data_idx++;
 	    is_image_data(data_idx) = true;
 	    parametric(data_idx) = false;
 	    have_cdata(data_idx) = false;
 
@@ -504,17 +504,17 @@ function __go_draw_axes__ (h, plot_strea
 	  endif
 
        case "patch"
          cmap = parent_figure_obj.colormap;
 	 [nr, nc] = size (obj.xdata);
 
 	 if (! isempty (obj.cdata))
 	   cdat = obj.cdata;
-	   if (strcmp (obj.cdatamapping, "direct"))
+	   if (strcmpi (obj.cdatamapping, "direct"))
 	     cdatadirect = true;
 	   endif
 	 else
 	   cdat = [];
 	 endif
 
 	 for i = 1:nc
 	   xcol = obj.xdata(:,i);
@@ -1197,17 +1197,17 @@ function __go_draw_axes__ (h, plot_strea
 	    fputs (plot_stream, sprintf(", %d \"#%02X%02X%02X\"", i - 1, 
 					col(1), col(2), col(3)));
 	  endif
 	endfor
 	fputs (plot_stream, ");\n");
       endif
     endif
 	    
-    if (strcmp (axis_obj.__colorbar__, "none"))
+    if (strcmpi (get (h, "__colorbar__"), "none"))
       fputs (plot_stream, "unset colorbox;\n");
     else
       ## FIXME If cbox_mirror is true we want to invert the tic labels
       ## but gnuplot doesn't allow that
       fputs (plot_stream, 
 	     sprintf ("set colorbox %s user origin %f,%f size %f,%f;\n",
 		      cbox_orient, cbox_origin, cbox_size));
     endif
